src/pages/index.js

```diff
@@ -15,39 +18,86 @@ const profile = document.querySelector(".profile");
 const profileEditButton = profile.querySelector(".profile__edit-button");
 const profileAddButton = document.querySelector(".profile__add-button");
 
+// const profile image edit buttton
+const profileImageButton = document.querySelector(".profile__image-edit");
+
 // all forms
 const forms = Array.from(document.forms);
 const formObjects = {};
 
+// hold card to delete
+let cardToDelete = null;
+
+const api = new API({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "3352b458-30c6-41be-9a42-00c52b81b385",
+    "Content-Type": "application/json",
+  },
+});
+
 // user info manipulation class
 const userInfo = new UserInfo({
   nameSelector: ".profile__name",
   jobSelector: ".profile__avocation",
+  photoSelector: ".profile__avatar",
 });
 
 // profile modal popup
 const profileModalPopup = new PopupWithForm({
   popupSelector: ".profile-modal",
   handleSubmitFunc: (data) => {
-    profileModalPopup.close();
-    userInfo.setUserInfo({
-      name: data.name,
-      job: data.job,
-    });
+    profileModalPopup.toggleSubmitText(true);
+    api
+      .patchProfileInformation({
+        name: data.name,
+        about: data.job,
+      })
+      .then((res) => {
+        profileModalPopup.close();
+        userInfo.setUserInfo({
+          name: data.name,
+          job: data.job,
+        });
+        profileModalPopup.toggleSubmitText(false);
```

Comment:
You need to return the **default** button text only in blocks `finally` so that the button could get the text back in any case: `success` or `failure`.

**Please, correct it everywhere**